lua/luasnip/util/environ.lua

```diff
@@ -19,6 +21,10 @@ local function _fill_eager_vars(env)
 	env.TM_CURRENT_WORD = util.word_under_cursor(cur, env.TM_CURRENT_LINE)
 	env.TM_LINE_INDEX = tostring(cur[1])
 	env.TM_LINE_NUMBER = tostring(cur[1] + 1)
+
+  env.SELECT_RAW = util.get_selection(util.SELECT_RAW)
+  env.SELECT_DEDENT = util.get_selection(util.SELECT_DEDENT)
+  env.TM_SELECTED_TEXT= util.get_selection(util.TM_SELECT)
```

Comment:
What do you think about keeping the function more like the older one eg. get and delete all selected at the same time? I don't think we need 3 calls for that.

Reply:
Sure, I'll do it, let me first finish with the worspaces thingy.

Reply:
Ah, nice :+1:

Reply:
I tried to do it the old way but doing  `env.var1, env.var2, env.var3 = get_selection()` breaks with the lazy loading, so either I keep it like this or I'll have to store the result in a table before passing it to each eager variable one by one.

Reply:
This is also fine, but I don't quite get what breaks by returning three values there?

Reply:
It breaks as if `env.var1, env.var2, env.var3`  calls  `__index` and then as the vars are not set yet, it fails.

Reply:
Oh okay, that I can believe.